### PR TITLE
Implement schema

### PR DESCRIPTION
**Pull request by talonchandler**
Implements the schema described in #421. 

@ziw-liu my apologies the first three commits are from different branches...I can't seem to find the right branch to merge this to.

### CODE REVIEW

**Comment by ziw-liu** [comment]
Why can't this target `main`?

**Comment by talonchandler** [comment]
Changed to target `main`. Sorry...when I started this branch `nm_retardance` hadn't merged, so I forgot to change it.

File: recOrder/calib/Calibration.py [hunk 1]
```diff
@@ -946,8 +946,8 @@ def write_metadata(self, notes=None):
                 }
             )
 
-        if not self.meta_file.endswith(".txt"):
-            self.meta_file += ".txt"
+        # if not self.meta_file.endswith(".txt"):
+        #    self.meta_file += ".txt"
```
Comment (ziw-liu) [inline recOrder/calib/Calibration.py:950]: Why do we need to keep this?

Comment (talonchandler) [reply recOrder/calib/Calibration.py]: Forgot to delete!

File: recOrder/acq/acquisition_workers.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import logging
 import os
```
Comment (ziw-liu) [inline recOrder/acq/acquisition_workers.py:4]: Unused import
```suggestion
```

File: recOrder/calib/calibration_workers.py [hunk 3]
```diff
@@ -5,14 +5,14 @@
 import logging
 import os
```
Comment (ziw-liu) [inline recOrder/calib/calibration_workers.py:6]: Unused import: glob, os, shutil

File: recOrder/io/utils.py [hunk 4]
```diff
@@ -13,6 +13,24 @@
...
 
 
+def add_index_to_path(path: Path):
+    """Takes a path to a file or folder and appends the smallest index that does
+    not already exist in that folder.
+
+    For example:
+    './output.txt' -> './output_0.txt' if no other files named './output*.txt' exist.
+    './output.txt' -> './output_2.txt' if './output_0.txt' and './output_1.txt' already exist.
+    """
```
Comment (ziw-liu) [inline recOrder/io/utils.py:23]: NumPy docs:
```suggestion
    Parameters
    ----------
    path: Path
        Base path to add index to

    Returns
    -------
    Path
    """
```

**Review by ziw-liu — APPROVED** [review]

**Pull request merged**